service-registry/src/main/java/org/apache/servicecomb/serviceregistry/client/IpPortManager.java

```diff
@@ -50,6 +50,11 @@
 
   private boolean autoDiscoveryInited = false;
 
+  public int getMaxRetryTimes() {
+    return currentAvailableIndex.get() < 2 ? 2 : currentAvailableIndex.get();
```

Comment:
This logic is not correct. Assume current index is 2, and failed, index becomes 0. In this situation, only retried 2 and you expect 3 times